br/pkg/lightning/mydump/region.go

```diff
@@ -303,11 +301,8 @@ func MakeSourceFileRegion(
 	rowIDMax := fileSize / divisor
 	// for compressed files, suggest the compress ratio is 1% to calculate the rowIDMax.
 	// set fileSize to INF to make sure compressed files can be read until EOF. Because we can't get the exact size of the compressed files.
-	// TODO: update progress bar calculation for compressed files.
 	if fi.FileMeta.Compression != CompressionNone {
-		// FIXME: this is not accurate. Need sample ratio in the future and use sampled ratio to compute rowIDMax
-		//  currently we use 500 here. It's a relatively large value for most data.
-		rowIDMax = fileSize * compressDataRatio / divisor
+		rowIDMax = fi.FileMeta.RealSize * CompressSizeFactor / divisor
```

Comment:
```
info.FileMeta.RealSize = int64(compressRatio * float64(info.FileMeta.FileSize))
```
The real size is the size of the de-compressed file, so we don't need to multiply the CompressSizeFactor anymore?

Reply:
lgtm